### PR TITLE
Remove vendored Regex

### LINKED ISSUE #8: Replace "regex" library usage
Currently, the git cmd uses "regex" functionality. Once the `ci` branch is merged, we want to remove the vendored regex functionality that depends on PRCRE2 and use something that has no external dependencies to make bootstrapping corral easier.

To that end

- Regex using code should be rewritten
- README should have "how to install pcre2" removed
- "vendored" regex library should be removed

### PR DESCRIPTION
**Pull request by rkallos**
Fixes #8

### CODE REVIEW

**Comment by SeanTAllen** [comment]
@rkallos what sort of tests would you run? it would make sense to start adding pony tests at this point and have a unit test for this functionality.

**Comment by rkallos** [comment]
So far the only code path that tests this is via `corral update`. It seems to work?
Running with my Franken-clone of pony-crdt:
```
FINE: tags for [1] github.com/jemc/pony-jason.git: 0
FINE: 
FINE: result err: 
FINE: tags for [0] github.com/mfelsche/ponycheck.git: 6
FINE:   tag:0.2.0
FINE:   tag:0.3.0
FINE:   tag:0.4.0
FINE:   tag:0.4.1
FINE:   tag:0.5.0
FINE:   tag:0.5.1
```

**Comment by cquinn** [comment]
I like this change.

**Comment by SeanTAllen** [comment]
@rkallos we have no tests so I'm good with you merging this whenever you are comfortable.

**Comment by rkallos** [comment]
I added the beginning of a test suite. As a result, I had to expose a `parse_tags` method that does the iteration over the lines from stdout. On the bright side, it looks like it works :)

File: corral/test/main.pony [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+      """
+    let expect: Array[String] =
+      ["0.2.0"; "0.3.0"; "0.4.0"; "0.4.1"; "0.5.0"; "0.5.1"]
+
+    let git = GitVcs(h.env)?
+    let rcv = {(res: Array[String] val) => None}
+    let res = GitQueryTags(git, consume rcv).parse_tags(stdout)
+    for pair in expect.pairs() do
+      (let i: USize, let expected: String) = pair
+        h.assert_eq[String](expected, res(i)?)
```
Comment (SeanTAllen) [inline corral/test/main.pony]: formatting appears off here

Comment (rkallos) [reply corral/test/main.pony]: Good catch!

**Comment by SeanTAllen** [comment]
Looks like I need to update settings to run CI for forked repos.

**Comment by SeanTAllen** [comment]
ok, ive turned on CI running for PRs from forked repos.

**Comment by rkallos** [comment]
Can you force a build on this branch, or should I push an empty commit to trigger it?

**Comment by EpicEric** [comment]
Nice!

I would suggest adding more lines with different formats that `git show-ref` outputs, such as this example from the ponyc repo:

```
b0b68f7a394ca6cd13198f2fdf7240bf7a116d8e refs/heads/master
329dda3c3db8f75c1186da9cf4fb05da183c389b refs/heads/2852-changelog
329dda3c3db8f75c1186da9cf4fb05da183c389b refs/remotes/origin/2852-changelog
b0b68f7a394ca6cd13198f2fdf7240bf7a116d8e refs/remotes/origin/HEAD
45a1d66ee94108f58ce5525e411c716871fcb37f refs/remotes/origin/MinGW64
cefc960800ba619e2a750fe49da06eb7408dc07b refs/remotes/origin/bionic-source-instructions
4d7087e672a3c056bbe2e01a54df3dcaa1b1bba5 refs/tags/0.1.0
652c628ed96dbdcdc36622c9b6e01981b0a066a7 refs/tags/0.1.1
4e26ed4fd55ce7f02a589d5dda4128e3c5003d55 refs/tags/0.1.2
d3f64c921b6d6355427caa46820f712cd4911928 refs/tags/0.1.3
41051c06abb768ecc041421adbbc192e76961c7c refs/tags/0.1.4
cb82ac111edb1a32d42fc8cf8fccbc77908a752a refs/tags/0.1.5
dcf27b504214ad30198c828ae4db2512862bed78 refs/tags/0.1.6
90989eb597967edc5ac6fca97c1b68cc2fc6b471 refs/tags/0.1.7
```

**Comment by rkallos** [comment]
Good idea! Done :)

**Comment by SeanTAllen** [comment]
@rkallos nice! test ran.

**Comment by rkallos** [comment]
Sweet. In that case, I think this is ready for merging. Any objections?

**Comment by SeanTAllen** [comment]
No objection from me.

**Comment by SeanTAllen** [comment]
Now that this is merged, I believe the nightly builds should start working.

**Pull request merged**